### ISSUE TITLE
Can't delete users

### ISSUE THREAD

**Issue by aabercrombie** [Tizra/Tizra-Customer-Tracker#306]
Trying to delete users from cmba.tizrapublisher.com. Get this error whenever trying to delete a page of users.

Request failed. Error sent back: >><!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html>
    <head>
        <title>Tizra AgilePublisher Error</title>
        <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
        <link id="adminCss" rel="stylesheet" type="text/css" href="/styles/admin/tizra.css" /> 
    </head>
    <body>

```
    <!-- ***************** -->
    <!-- Begin Banner -->
    <!-- ***************** -->
    <div id="banner">
        <div class="banner-title">
            <span class="logotype">Tizra AgilePublisher<sup>TM</sup></span>
        </div>
    </div>
    <!-- End Banner -->

    <!-- ***************** -->
    <!-- Begin Content Area -->
    <!-- ***************** -->
    <div id="content">
        <p>
```

The web page you seek<br />Cannot be located, but<br />Countless more exist</p>
        <p><a href="/">Return to site home</a></p>
        </div>
        
    </body>
</html>

**Comment by daviddurand** [comment]
I'll work on making the error message for this kind of problem more informative. The logs show there is a problem deleting one of the accounts, which was a database issue.

There was an account starting with 2ggoldberg that was causing the problem. I cleaned it up. Could have been from a loading error long ago.

**Comment by aabercrombie** [comment]
I just tried again this morning and ran into the same error. (Set list length of users to 1000, checked all, tried to delete, got to 100% and then got error message.)

**Comment by aabercrombie** [comment]
If it helps any, every time I try to leave that page after doing something I also get an alert that what I did wasn't saved and am I sure I want to leave the page. Seeing as no changes had been made, maybe something else is going on?

**Comment by daviddurand** [comment]
There must be some number of bad users in the lot. Perhaps a failed load at some point.

Deletion isn't a "safe" operation, so I can't find the problem accounts, because I can't start deleting accounts randomly on a test basis.

You don't have to delete the accounts before a load, as they will be updated as necessary during the load. The import will delete any tags that are no longer attached to an account, which should remove them from any access groups that they are no longer entitled to.

There are other potential workarounds: 
- You could just mark the relevant accounts as expired. In that case they won't be able to log in, which may accomplish what you want.
- You could use a new set of tags and change the definitions of the user sets that license content. Then the new load will grant access to the right people, and the old accounts (with the old tags) won't have access even if they can still log in.

If you really need to delete the accounts, you'll have to delete in smaller batches and close in on at least a few accounts that can't be deleted, so that I can see if there is a systematic way to find and fix them.  I made and deleted an account without a problem, so it's not a general problem with deleting accounts.

**Comment by daviddurand** [comment]
The change error is just a warning that the operation was not completed, before you lose your checkboxes -- you can ignore it.

**Comment by aabercrombie** [comment]
No matter what users I try to delete from that main page, I get the error. I tried random accounts on all pages. No matter what I try I get that error. If it's only a few accounts here and there, that doesn't make sense. 

When I tried to upload users with new tags yesterday, anyone who was already in the system didn't have their tags updated, they were just skipped. So the system seems to also not be updating the tags properly.

The problem is we don't know which accounts are relevant anymore. Since it's close to impossible to get the system to download a spreadsheet of all 5000 users in the system (it keeps timing out on me, that was what I tried first), we don't know who's current and who's not on the new member list, so it would be easiest to delete everyone and start over.
